wgpu-hal/examples/ray-traced-triangle/main.rs

```diff
@@ -0,0 +1,1124 @@
+                0,
+                0xff,
+                0,
+                0,
+                unsafe { device.get_acceleration_structure_device_address(&blas) },
+            ),
+            AccelerationStructureInstance::new(
+                &Affine3A::from_translation(Vec3 {
+                    x: -1.0,
+                    y: -1.0,
+                    z: -2.0,
+                }),
+                0,
+                0xff,
+                0,
+                0,
+                unsafe { device.get_acceleration_structure_device_address(&blas) },
+            ),
+            AccelerationStructureInstance::new(
+                &Affine3A::from_translation(Vec3 {
+                    x: 1.0,
+                    y: -1.0,
+                    z: -2.0,
+                }),
+                0,
+                0xff,
+                0,
+                0,
+                unsafe { device.get_acceleration_structure_device_address(&blas) },
+            ),
+        ];
+
+        let instances_buffer_size =
+            instances.len() * std::mem::size_of::<AccelerationStructureInstance>();
+
+        let instances_buffer = unsafe {
+            let instances_buffer = device
+                .create_buffer(&hal::BufferDescriptor {
+                    label: Some("instances_buffer"),
+                    size: instances_buffer_size as u64,
+                    usage: hal::BufferUses::MAP_WRITE
+                        | hal::BufferUses::TOP_LEVEL_ACCELERATION_STRUCTURE_INPUT,
+                    memory_flags: hal::MemoryFlags::TRANSIENT | hal::MemoryFlags::PREFER_COHERENT,
+                })
+                .unwrap();
+
+            let mapping = device
+                .map_buffer(&instances_buffer, 0..instances_buffer_size as u64)
+                .unwrap();
+            ptr::copy_nonoverlapping(
+                instances.as_ptr() as *const u8,
+                mapping.ptr.as_ptr(),
+                instances_buffer_size,
+            );
+            device.unmap_buffer(&instances_buffer).unwrap();
+            assert!(mapping.is_coherent);
+
+            instances_buffer
+        };
+
+        if let hal::AccelerationStructureEntries::Instances(ref mut i) = tlas_entries {
+            i.buffer = Some(&instances_buffer);
+            assert!(
+                instances.len() <= i.count as usize,
+                "Tlas allocation to small"
+            );
+        }
+
+        let cmd_encoder_desc = hal::CommandEncoderDescriptor {
+            label: None,
+            queue: &queue,
+        };
+        let mut cmd_encoder = unsafe { device.create_command_encoder(&cmd_encoder_desc).unwrap() };
+
+        unsafe { cmd_encoder.begin_encoding(Some("init")).unwrap() };
+
+        unsafe {
+            cmd_encoder.place_acceleration_structure_barrier(hal::AccelerationStructureBarrier {
+                usage: hal::AccelerationStructureUses::empty()
+                    ..hal::AccelerationStructureUses::BUILD_OUTPUT,
+            });
```

Comment:
Why isn't this just a normal barrier?

Reply:
As far as I can tell Vulkan and DX12 require quite different syncronisation, (since dx uses a buffer for the acceleration structure while vulkan has its own resource type).
So this function is in preperation of the DX12 implementation.